### DISCUSSION TITLE
Regras de negocio (Entities ou Commands)

### DISCUSSION THREAD

**Discussion by igorRomulo** [balta-io/7196#45] [Perguntas e Respostas] [answered]
@andrebaltieri estou com duvida sobre regras de negocio, seguinte vamos supor que tenho uma aplicação que tem fazer alguns calculos, onde que os mesmos deveriam fica no dentro das entities ou dos commands?, penso que dentro das entity e os commands valida apenas as entradas e saidas de dados

**Comment by andrebaltieri** [comment, marked as answer]
@igorRomulo a localização das regras depende da arquitetura que está usando... mas nos DTOs já te adianto que não são um bom lugar normalmente.

Eles podem (E devem) fazer as validações dos dados, mas as regras de negócio ou ficam nas entidades ou o mais próximo possível delas.
